File: packages/android_intent_plus/lib/src/put_classes/put_double.dart
import 'package:android_intent_plus/src/put_classes/base/put_base.dart';

class PutDouble extends PutBase<double> {
  PutDouble({required String key, required double value})
      : super(key: key, value: value);

  @override
  String get javaClass => 'PutDouble';

  factory PutDouble.fromJson({required String key, required dynamic value}) {
    return PutDouble(key: key, value: value as double);
  }

  @override
  Map<String, dynamic> toJson() {
    return <String, dynamic>{
      'key': key,
      'javaClass': javaClass,
      'value': value,
    };
  }
}

File: packages/android_intent_plus/android/src/main/java/dev/fluttercommunity/plus/androidintent/Bundle/PutClasses/PutDouble.java
package dev.fluttercommunity.plus.androidintent.Bundle.PutClasses;

import android.os.Bundle;

import org.json.JSONException;
import org.json.JSONObject;

import dev.fluttercommunity.plus.androidintent.Bundle.Constants;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.base.PutBase;


public class PutDouble extends PutBase<Double> {

  public static final String JAVA_CLASS = "PutDouble";

  public PutDouble(String key, Double value) {
    super(key, JAVA_CLASS, value);
  }

  public static void convert(Bundle bundle, PutDouble putDouble) {
    bundle.putDouble(putDouble.key, putDouble.value);
  }

  public static PutDouble fromJson(JSONObject jsonObject) throws JSONException {
    return new PutDouble(jsonObject.getString(Constants.KEY), jsonObject.getDouble(Constants.VALUE));
  }

  @Override
  public JSONObject toJson() throws JSONException {
    JSONObject jsonObject = new JSONObject();
    jsonObject.put("key", key);
    jsonObject.put("javaClass", JAVA_CLASS);
    jsonObject.put("value", value);
    return jsonObject;
  }
}
